moses/atomese/representation/atomese_representation.cc

```diff
@@ -0,0 +1,444 @@
+/** atomese_representation.cc ---
+ *
+ * Copyright (C) 2018 OpenCog Foundation
+ *
+ * Author: Yidnekachew Wondimu <searchyidne@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <moses/comboreduct/table/table_io.h>
+#include <opencog/util/comprehension.h>
+#include <opencog/atoms/base/Link.h>
+#include <opencog/atoms/base/Node.h>
+#include <opencog/atoms/core/NumberNode.h>
+#include <opencog/truthvalue/SimpleTruthValue.h>
+#include "atomese_representation.h"
+
+namespace opencog {
+namespace atomese {
+
+using namespace std;
+using namespace combo;
+
+/**
+ * Represent feature labels as a list of schema nodes.
+ *
+ * @param labels feature labels
+ * @return
+ */
+Handle parse_header(const vector<string>& labels)
+{
+	HandleSeq cols;
+	for (string str : labels)
+		cols.push_back(createNode(SCHEMA_NODE, str));
+
+	return createLink(cols, LIST_LINK);
+}
+
+/**
+ * Create a node representing a row having
+ * the row number as its value.
+ *
+ * @param row_num row number
+ * @return
+ */
+Handle mk_row_number_cell(const int& row_num)
+{
+	return createNode(NODE, "r" + to_string(row_num));
+}
+
+/**
+ * Create a NumberNode given a contin value.
+ *
+ * @param contin contin value of a cell
+ * @return
+ */
+Handle mk_real_cell(const contin_t& contin)
+{
+	return createNumberNode(contin)->get_handle();
+}
+
+/**
+ * Represent a real cell as an Execution of the corresponding feature
+ * on its row (i.e Schema of feature on Node of row).
+ *
+ * @param contin contin value of a cell
+ * @param row_num row number
+ * @param label label of the feature
+ * @return
+ */
+Handle mk_real_cell_exec(const contin_t& contin,
+                         const int& row_num, const string& label)
+{
+	return createLink(EXECUTION_LINK,
+	                  createNode(SCHEMA_NODE, label),
+	                  mk_row_number_cell(row_num),
+	                  mk_real_cell(contin));
+}
+
+/**
+ * Create a True / False link based on a given builtin value.
+ *
+ * @param b builtin value
+ * @return
+ */
+Handle mk_boolean_cell(const builtin& b)
```

Comment:
Same remark about builtin ref passing.

Reply:
`builtin` is an enum. Shouldn't we pass it by ref?